src/shogun/regression/LeastAngleRegressionImpl.cpp

```diff
@@ -40,23 +40,23 @@ using namespace Eigen;
 
 namespace shogun
 {
-SGMatrix<float64_t> CLeastAngleRegressionImpl::cholesky_insert_helper(SGMatrix<float64_t> X, SGMatrix<float64_t> R, VectorXd x_vec, SGVector<float64_t> active_set)
+SGMatrix<float64_t> CLeastAngleRegressionImpl::cholesky_insert_helper(Ref<MatrixXd> map_X, SGMatrix<float64_t>& R, int32_t i_max_corr, SGVector<float64_t> active_set)
```

Comment:
Leads to additional overhead if I save the transpose. If I use `map.transpose().col(i)` too many cache misses. So options might be pass Eigen things or manually tranpose SGMatrix initally or any suggestions ?